lib/tower_cli/models/base.py

```diff
@@ -829,11 +830,158 @@ class MonitorableResource(ResourceMethods):
+        """
+        Print out the standard out of a unified job to the command line.
+        For Projects, print the standard out of most recent update
+        For Inventory Sources, print standard out of most recent sync
+        For Jobs, print the job's standard out
+        """
+        # resource is Unified Job Template
+        if self.unified_job_type != self.endpoint:
+            unified_job = self.last_job_data(pk, **kwargs)
+            pk = unified_job['id']
+        # resource is Unified Job, but pk not given
+        elif not pk:
+            unified_job = self.get(**kwargs)
+            pk = unified_job['id']
+
+        content = self.lookup_stdout(pk, start_line, end_line)
+        if len(content) > 0:
+            click.echo(content, nl=0)
+
+        return {"changed": False}
+
+    @resources.command
+    @click.option('--interval',
+                  default=1, help='Polling interval to refresh content '
+                                  'from Tower.')
+    @click.option('--timeout', required=False, type=int,
+                  help='If provided, this command (not the job) will time out '
+                       'after the given number of seconds.')
+    @click.option('--stdout', is_flag=True,
+                  help='Prints stdout on a rolling basis.')
+    def monitor(self, pk, parent_pk=None, interval=0.01, timeout=None,
+                outfile=sys.stdout, **kwargs):
+        """
+        Stream the standard output from a
+            job, project update, or inventory udpate.
+        """
+        # If we do not have the unified job info, infer it from parent
+        if pk is None:
+            pk = self.last_job_data(parent_pk, **kwargs)['id']
+        job_endpoint = '%s%s/' % (self.unified_job_type, pk)
+
+        # Pause until job is in running state
+        self.wait(pk, exit_on='running')
+
+        # Loop initialization
+        start = time.time()
+        start_line = 0
+        result = client.get(job_endpoint).json()
+
+        click.echo('\033[0;91m------Starting Standard Out Stream------\033[0m',
+                   nl=2, file=outfile)
+
+        # Poll the Ansible Tower instance for status and content,
+        # and print standard out to the out file
+        while result['status'] != 'successful':
+
+            # Polling loop exit conditions
+            if result['failed']:
+                raise exc.JobFailure('Job failed.')
+            elif timeout and time.time() - start > timeout:
+                raise exc.Timeout('Monitoring aborted due to timeout.')
+
+            # Put the process to sleep briefly.
+            time.sleep(interval)
+
+            # Make request to get standard out
+            content = self.lookup_stdout(pk, start_line)
+
+            # In the first moments of running the job, the standard out
+            # may not be available yet
+            if not content.startswith("Waiting for results"):
+                line_count = content.count('\n')
```

Comment:
I'll try finding some. In the mean time, strings are equipped with a built-in method `splitlines` that could be helpful.